src/components/App/index.js

```diff
@@ -1,19 +1,25 @@
 import React from "react";
-
 import Routing from "../../routing";
 import Header from "../Header";
+import Footer from "../Footer";
+// import "./styles/styles.css";
+import { ThemeProvider } from "@mui/material/styles";
+import theme from "../../theme/theme";
 
 import "./styles.scss";
 
 function App() {
   return (
```

Comment:
As per @kartiks26 's suggestion to use `useRouteMatch`, which is definitely the way to go:
```suggestion
  const showHeader = !useRouteMatch({
    path: HideHeader
  });
  const showFooter = !useRouteMatch({
    path: HideFooter
  });
  return (
```